api/local/setProgress.ts

```diff
@@ -0,0 +1,8 @@
+export default async function setProgress(progress: string) {
+  try {
+    localStorage.setItem('SavingSatoshiProgress', JSON.stringify(progress))
```

Comment:
Not sure why you are using `JSON.stringify` here, this will result in extra `"` characters around the lessonId. This in turn forces you to use `.replace(/['"]/g, '')` [here](https://github.com/saving-satoshi/saving-satoshi/pull/399/files#diff-1a2ca889da37ca6ad47d64eaae05a834b5886e6a8bba5b1c2ea555045c97dccdR7).

Reply:
Not using Json.stringify() causes the following error.

<img width="1503" alt="image" src="https://user-images.githubusercontent.com/85434418/234794581-64c330a3-c85f-4b16-80c1-d40659ecc9a3.png">

That's why using `JSON.stringify()` while saving and using `.replace(/['"]/g, '')` while reading is the cleanest solution in our case.

Reply:
Hey @shaavan I checked out your branch locally, made the changes I suggested and was unable to reproduce your issue. 

The destructuring error happens if you try to do something like this:
```ts
const { path } = undefined
```
Which I dont see you doing anywhere in your code, perhaps minification has changed the names of variables but I'm wondering why you're seeing minified code in your browser window. Can you provide some steps to reproduce?

Reply:
> Can you provide some steps to reproduce?

1. I made the changes you suggested. (Same as bd82a76b651ee822a133fb0b90dd916a2fcec929)
2. Compiled and ran the branch (`npm run build` `npm run start`)
3. I navigated to [Intro-1 page](http://localhost:3000/chapters/chapter-1/intro-1) and noticed this error.